changelog_unreleased/api/17331.md

```diff
@@ -0,0 +1,15 @@
+#### Add `isSupported` function support for `languages` API (#17331 by @JounQin)
+
+Previously, `languages` API for custom plugin only supported to infer parser based on the file basename or extension.
+
+Prettier main added `isSupported: (file: string) => boolean` function to allow plugin check if file is supported based on the full path (eg: files in a specific directory), the `file` parameter could be a normal path or a url string like `file:///C:/test.txt`.
```

Comment:
@sosukesuzuki How about:

```suggestion
Prettier main added `isSupported: (file: string) => boolean` function to allow plugin check if file is supported based on the full path (eg: files in a specific directory), the `file` parameter could be a normal path or a url string like `file:///C:/test.txt`.

If no `isSupported` provided, it just behaviors the smae way as before.
```

Reply:
`smae` -> `same`

Reply:
> `smae` -> `same`

Suggestion changed.

Reply:
Should I just commit the suggestion? @fisker @sosukesuzuki